### PR TITLE
[APB-2056][AO] detailed monitoring of subscription journey

### PR DESCRIPTION
**Pull request by arturopala**

### CODE REVIEW

File: app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SubscriptionController.scala [hunk 1]
```diff
@@ -96,12 +98,14 @@ class SubscriptionController @Inject()
...
       hasEnrolments match {
         case true => Future(Redirect(routes.CheckAgencyController.showHasOtherEnrolments()))
-        case false => sessionStoreService.fetchKnownFactsResult.map(_.map { knownFactsResult =>
-          Ok(html.subscription_details(knownFactsResult.taxpayerName, initialDetailsForm.fill(
-            InitialDetails(knownFactsResult.utr, knownFactsResult.postcode, null, null, null))))
-        }.getOrElse {
-          sessionMissingRedirect()
-        })
+        case false =>
+          mark("Count-Subscription-CleanCreds-Success")
```
Comment (arkowalski) [inline app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SubscriptionController.scala]: why only mark() false?, mark "true" above and the "getOrElse" below also?

File: app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SubscriptionController.scala [hunk 2]
```diff
@@ -159,7 +173,10 @@ class SubscriptionController @Inject()
...
               desAddressForm.bindAddressLookupFrontendAddress(details.utr, address).fold(
                 formWithErrors => Future successful Ok(html.address_form_with_errors(formWithErrors)),
-                validDesAddress => subscribe(details, validDesAddress).map(redirectSubscriptionResponse)
+                validDesAddress => {
+                  mark("Count-Subscription-AddressLookup-Success")
+                  subscribe(details, validDesAddress).map(redirectSubscriptionResponse)
+                }
               )
             }
           }.getOrElse(Future.successful(sessionMissingRedirect()))
```
Comment (arkowalski) [inline app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SubscriptionController.scala]: mark() ?

File: app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SubscriptionController.scala (hunk 1 above)
Comment (arturopala) [reply app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SubscriptionController.scala]: we are only interested in key steps, not in all events

File: app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SubscriptionController.scala (hunk 2 above)
Comment (arturopala) [reply app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SubscriptionController.scala]: as above, we aim to monitor only key points

**Pull request merged**